diff --git a/freqtrade/optimize/backtesting.py b/freqtrade/optimize/backtesting.py
index 493c7567f..8a3ec7037 100644
--- a/freqtrade/optimize/backtesting.py
+++ b/freqtrade/optimize/backtesting.py
@@ -413,6 +413,7 @@ class Backtesting:
         """
         Get close rate for backtesting result
         """
+        return row[OPEN_IDX]
         # Special handling if high or low hit STOP_LOSS or ROI
         if exit.exit_type in (
                 ExitType.STOP_LOSS, ExitType.TRAILING_STOP_LOSS, ExitType.LIQUIDATION):
